packages/core/src/RenderPipeline/MaskManager.ts

```diff
@@ -63,6 +64,25 @@ export class MaskManager {
     }
   }
 
+  clear(engine: Engine, renderQueueType: RenderQueueType, pipelineStageTagValue: string): void {
+    const incrementMaskQueue = MaskManager.getMaskIncrementRenderQueue();
+    incrementMaskQueue.renderQueueType = renderQueueType;
+    incrementMaskQueue.clear();
+    const decrementMaskQueue = MaskManager.getMaskDecrementRenderQueue();
+    decrementMaskQueue.renderQueueType = renderQueueType;
+    decrementMaskQueue.clear();
+
+    this.buildMaskRenderElement(0, incrementMaskQueue, decrementMaskQueue);
+
+    const { _renderContext: context, _batcherManager: batcherManager } = engine;
+    incrementMaskQueue.batch(engine._batcherManager);
+    batcherManager.uploadBuffer();
+    incrementMaskQueue.render(context, pipelineStageTagValue, StencilOperation.IncrementSaturate);
+    decrementMaskQueue.batch(engine._batcherManager);
+    batcherManager.uploadBuffer();
+    decrementMaskQueue.render(context, pipelineStageTagValue, StencilOperation.DecrementSaturate);
```

Comment:
The uploadBuffer and batch here are used in many places and can be encapsulated into a function.

Reply:
yeap~